compiler/rustc_passes/src/liveness.rs

```diff
@@ -274,9 +270,8 @@ impl IrMaps<'tcx> {
     }
 
     fn add_live_node(&mut self, lnk: LiveNodeKind) -> LiveNode {
-        let ln = LiveNode(self.num_live_nodes as u32);
+        let ln = LiveNode(self.lnks.len() as u32);
```

Comment:
can we instead use `rustc_newtype_index` for `LifeNode` and `Variable`?

This changes these methods to simply `self.lnks.push(lnk)`

Reply:
I used newtype_index and IndexVec in a few places. I will take a look if INVALID_NODE can be replaced with an Option, especially that newtype_index has a niche for that purpose, but that would be another PR.